android/engine/src/main/java/org/smartregister/fhircore/engine/cql/LibraryEvaluator.kt

```diff
@@ -193,39 +197,63 @@ class LibraryEvaluator @Inject constructor() {
     return auxPatientDataObj.toString()
   }
 
-  fun runCqlLibrary(
-    library: org.hl7.fhir.r4.model.Library,
-    helper: org.hl7.fhir.r4.model.Library,
-    valueSet: Bundle,
-    data: Bundle
+  suspend fun runCqlLibrary(
+    libraryId: String,
+    patient: Patient,
+    resources: List<Resource>,
+    // TODO refactor class by modular and single responsibility principle
```

Comment:
should we do this refactor now?

Reply:
This would require refactor of whole class which includes the CQL for ANC as well. Since this impacting the other work we doing for quest for multiple tickets hence I did not touch the parts which can lead to massive changes. I can revisit this after finishing the current tickets as quest or this method specifically might not need massive changes rather it would be ANC module